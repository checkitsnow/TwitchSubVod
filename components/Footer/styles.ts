import styled from 'styled-components';
import { appearFromTop } from '@/utils/animations/appearFromTop';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
  position: sticky;
  top: 100%;
  padding: 1rem;
  margin-bottom: 5rem;

  animation: ${appearFromTop} 0.5s ease-in;

  .kofi-img {
    border: 0px;
    height: 2.5rem;
  }

  a {
    color: var(--text);
    font-weight: 700;
  }

  span {
    position: relative;
    display: flex;
    padding: 1rem;
  }
`;
