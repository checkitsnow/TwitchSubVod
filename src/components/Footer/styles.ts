import styled, { keyframes } from 'styled-components';

export const appearFromTop = keyframes`
from {
    opacity: 0;
    transform: translateY(-1rem);
  }
  to {
    opacity: 1;
    transform: translateY(0);
  }
`;

export const Container = styled.div`
  display: flex;
  position: sticky;
  top: 100%;
  padding: 1rem;

  animation: ${appearFromTop} 0.5s ease-in;

  a {
    color: var(--text);
    font-weight: 700;
  }
`;
