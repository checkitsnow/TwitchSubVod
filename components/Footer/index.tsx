import React from 'react';

import { Container } from './styles';

const Footer: React.FC = () => {
  return (
    <Container>
      <a
        href="https://ko-fi.com/pogulive"
        target="_blank"
        rel="noopener noreferrer"
      >
        <img
          className="kofi-img"
          src="https://cdn.ko-fi.com/cdn/kofi5.png?v=2"
          alt="Buy Me a Coffee at ko-fi.com"
        />
      </a>
      <a
        href="https://twitter.com/pogulive"
        target="_blank"
        rel="noopener noreferrer"
      >
        <img
          src="https://abs.twimg.com/favicons/twitter.ico"
          alt="twitter logo"
        />
      </a>
    </Container>
  );
};

export default Footer;
