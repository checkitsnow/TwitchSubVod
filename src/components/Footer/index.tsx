import React from 'react';

import { Container } from './styles';

const Footer: React.FC = () => {
  return (
    <Container>
      <span>
        This project is in no way affiliated with Twitch Interactive, Inc.
        Amazon.com, Inc., or any company mentioned on this page.
        <a href="https://github.com/Alissonsleal/TwitchSubVod">
          View Source code
        </a>
      </span>
    </Container>
  );
};

export default Footer;
