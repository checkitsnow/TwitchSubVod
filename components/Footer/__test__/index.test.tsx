import { render } from '@testing-library/react';
import Footer from '..';

it('should render Footer correctly', () => {
  const tree = render(<Footer />);
  expect(tree).toMatchSnapshot();
});
