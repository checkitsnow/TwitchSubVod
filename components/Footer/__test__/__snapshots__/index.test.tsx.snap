// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Footer correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="styles__Container-w3v8va-0 kVLkFL"
      >
        <a
          href="https://ko-fi.com/pogulive"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Buy Me a Coffee at ko-fi.com"
            class="kofi-img"
            src="https://cdn.ko-fi.com/cdn/kofi5.png?v=2"
          />
        </a>
        <a
          href="https://twitter.com/pogulive"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="twitter logo"
            src="https://abs.twimg.com/favicons/twitter.ico"
          />
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="styles__Container-w3v8va-0 kVLkFL"
    >
      <a
        href="https://ko-fi.com/pogulive"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Buy Me a Coffee at ko-fi.com"
          class="kofi-img"
          src="https://cdn.ko-fi.com/cdn/kofi5.png?v=2"
        />
      </a>
      <a
        href="https://twitter.com/pogulive"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="twitter logo"
          src="https://abs.twimg.com/favicons/twitter.ico"
        />
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
